support-frontend/assets/pages/subscriptions-redemption/components/redemptionForm.tsx

```diff
@@ -29,142 +28,162 @@ import type {
-	return (
-		<div role="status" aria-live="assertive" css={mainCss}>
-			<InlineError>
-				We notice you&apos;ve already set up a Guardian username and password,
-				please complete your account to proceed.
-			</InlineError>
-			<div css={detailsCss}>
-				<ol>
-					<li>
-						Go to{' '}
-						<a
-							href="https://manage.theguardian.com/account-settings"
-							target="_blank"
-							rel="noopener noreferrer"
-						>
-							Account settings
-						</a>
-						, add your first name and last name and save changes.
-					</li>
-					<li>Refresh this page to continue</li>
-				</ol>
-			</div>
-		</div>
-	);
-};
+const connector = connect(mapStateToProps, mapDispatchToProps());
+
+type PropTypes = ConnectedProps<typeof connector>;
+
+const redemptionForm = css`
+	padding: ${space[2]}px;
+`;
+
+const instructionsDivCss = css`
+	margin-top: -10px;
+	padding: ${space[2]}px;
+	${from.tablet} {
+		min-height: 475px;
+	}
+	hr {
+		border: 0;
+		border-top: solid 1px ${line.primary};
+	}
+`;
+
+const hrCss = css`
+	margin-bottom: 16px;
+`;
+const headingCss = css`
+	${headline.xsmall()};
+	font-weight: bold;
+	margin-bottom: 16px;
+`;
 
 function RedemptionForm(props: PropTypes) {
-	const mainCss = css`
-		padding: ${space[2]}px;
-	`;
-	const instructionsDivCss = css`
-		margin-top: -10px;
-		padding: ${space[2]}px;
-		${from.tablet} {
-			min-height: 475px;
-		}
-		hr {
-			border: 0;
-			border-top: solid 1px ${line.primary};
-		}
-	`;
-	const hrCss = css`
-		margin-bottom: 16px;
-	`;
-	const headingCss = css`
-		${headline.xsmall()};
-		font-weight: bold;
-		margin-bottom: 16px;
-	`;
 	const validationText = props.error ? null : 'This code is valid';
-	const signedIn = doesUserAppearToBeSignedIn();
-	const missingNames =
-		signedIn && props.user.firstName === '' && props.user.lastName === '';
-	const buttonText = signedIn ? 'Activate' : 'Continue to account setup';
+
 	return (
 		<div>
 			<Content>
 				<CheckoutLayout wrapPosition="bottom" aside={<ProductSummary />}>
 					<Form
 						onSubmit={(ev) => {
 							ev.preventDefault();
+							props.submitForm();
 						}}
 					>
-						<div css={mainCss}>
-							<h2 css={headingCss}>
+						<div css={redemptionForm}>
+							<h1 css={headingCss}>
 								Enjoy your Digital Subscription from the Guardian
-							</h2>
-							{missingNames ? <MissingNamesMessage /> : null}
-							<div>
+							</h1>
+							<FormSection>
 								<TextInput
 									autoComplete="off"
-									value={props.userCode}
+									value={props.userCode ?? ''}
 									onChange={(e) => props.setUserCode(e.target.value)}
-									error={props.error}
-									success={validationText}
+									error={props.error ?? ''}
+									success={validationText ?? ''}
 									label="Insert code"
-									css={css`
-										max-width: 300px;
-									`}
 								/>
-							</div>
+							</FormSection>
+							<FormSection title="Your details">
+								<PersonalDetails
+									firstName={props.firstName}
```

Comment:
Github is really not helping out here is it?

Reply:
I'm not sure what you're seeing, everything looks normal to me.

Reply:
Everything is wrapping in a weird way, it's fine though - just need to squint a bit